### PR TITLE
Silent audio, 14.5 update and CallProvider callbacks

### PR DESCRIPTION
**Pull request by jptaylor**
- Updated iOS target to >14.5 to remove the need for a hosted bridge file
- Correctly remove participant audio when leaving a meeting
- Implemented silent audio to prevent iOS switching devices on mute
- Correctly implemented effect deps to unblock ESLint / Netlify deploy
- Added effect that resumes audio tracks on Safari when device changes
- Various event flow optimisations

### CODE REVIEW

**Review by regexident — APPROVED** [review]
iOS-related stuff LGTM

File: react/README.md [hunk 1]
```diff
@@ -11,19 +11,21 @@ To run locally:
...
 
-Once you've deployed your own server, make sure to update the necessary URLs as noted in: 
+Once you've deployed your own server, make sure to update the necessary URLs as noted in:
 
 - react/src/CallProvider.jsx
-- react/public/ios-bridge.html
 
+## A note for Safari users
+
+Safari prevents the auto playing of audio by default. You can enable this in Preferences > Websites > Auto-Play
```
Comment (kimberleejohnson) [inline react/README.md:31]: I defer to you on whether or not it's also worth mentioning how we had to go back and "Unmute" multiple times.

**Review by kimberleejohnson — APPROVED** [review]
Looks great to me! Thank you for all this work. I left a comment but I leave adding that to the README or not up to you!

**Review by kompfner — APPROVED** [review]
👍

**Comment by kompfner** [comment]
cool! should we switch to `persistentTrack` in the ios bridge as well?

**Comment by kompfner** [comment]
the silent audio track also helped on web, too? what problem does it solve there?

**Comment by jptaylor** [comment]
> the silent audio track also helped on web, too? what problem does it solve there?

It didn't - I was experimenting to see if mobile Safari was fixed with it in place (it made no difference due to autoplay)

**Pull request merged**